frame/contracts/src/exec.rs

```diff
@@ -356,7 +356,15 @@ where
 		// Assumption: pay_rent doesn't collide with overlay because
 		// pay_rent will be done on first call and dest contract and balance
 		// cannot be changed before the first call
-		let contract_info = rent::pay_rent::<T>(&dest);
+		let (rent_outcome, contract_info) = rent::pay_rent::<T>(&dest);
+
+		// Deposit an event to indicate contract eviction.
+		if rent_outcome == rent::RentOutcome::Evicted {
+			self.deferred.push(DeferredAction::DepositEvent {
+				event: RawEvent::Evicted(dest.clone()),
+				topics: Vec::new(),
+			});
+		}
```

Comment:
but you can't used deferred field to store the Evicted event because the removal of a contract when it is called happens directly, not in the overlay.

i.e. if contract A call B and B call C and C is evicted and B trap with error then the deferred values are not merged to A.

But what you can simply do is actually make the deposit of the event in pay_rent as it is direct eviction.

Reply:
Did the requested changes and also noticed that it might be handy if the event would provide information about whether a tombstone has been created or not.